Api fix (#518)
smartcontractkit/documentation: docs/Using Any API/request-and-receive-data.md

---
![Chainlink Any API](/files/8c35025-Request__Receive_Data.png)

# Overview

Chainlink enables your contracts to access to *any* external data source through our decentralized oracle network.

Whether your contract requires sports results, the latest weather, or any other publicly available data, the [Chainlink contract library](https://github.com/smartcontractkit/chainlink/tree/master/contracts) provides the tools required for your contract to consume it.

> ⚠️ Note on Price Feed Data
> 
> If your smart contracts need access to price feed data, try using [Chainlink Data Feeds](../using-chainlink-reference-contracts/).

# Getting Started

We understand making smart contracts compatible with off-chain data adds to the complexity of building smart contracts. We created a framework with minimal requirements, yet unbounded flexibility, so developers can focus more on the functionality of smart contracts rather than what feeds them.

Chainlink’s decentralized oracle network provides smart contracts with the ability to push and pull data, facilitating the interoperability between on-chain and off-chain applications.

## Requesting Off-chain Data

> Outlined below are multiple ways developers can connect smart contracts to off-chain data feeds. Click on each request type to learn more:

Reviewer: ```suggestion
Outlined below are multiple ways developers can connect smart contracts to off-chain data feeds. Click a request type to learn more about it:
```
